ansible/library/test_facts.py

```diff
@@ -0,0 +1,187 @@
+            "vms1-1": {
+                "dut": "str-s6000-1",
+                "owner": "Tests vms",
+                "ptf_image_name": "docker-ptf-sai-mlnx",
+                "ptf_ip": "10.255.0.178",
+                "ptf_netmask": "255.255.255.0",
+                "server": "server_1",
+                "testbed-name": "vmst-1",
+                "topo": "t1",
+                "vm_base": "VM0100"
+            }
+            ....
+        }
+        "topo_testcases": {
+                ptf32:     [arpall, copp, everflow, neighbour, neighbour_mac_noptf, qos, snmp, syslog]
+                ptf64:     [arpall, copp, everflow, neighbour, neighbour_mac_noptf, qos, snmp, syslop]
+                t0:        [bgp_fact, bgp_speaker, decap, dhcp_relay, fast-reboot, fib, fdb, lldp, lag_2, pfc_wd]
+                t0-64:     [bgp_fact, bgp_speaker, decap, dhcp_relay, fast-reboot, fib, fdb, lldp, lag_2, pfc_wd]
+                t0-64-32:  [bgp_fact, bgp_speaker, decap, dhcp_relay, fast-reboot, fib, fdb, lldp, lag_2, pfc_wd]
+                t1:        [acl, bgp_fact, bgp_multipath_relax, decap, everflow_testbed, fib, lldp, pfc_wd]
+                t1-lag:    [acl, bgp_fact, bgp_multipath_relax, decap, everflow_testbed, fib, lldp, lag_2, pfc_wd]
+                t1-64-lag: [acl, bgp_fact, bgp_multipath_relax, decap, everflow_testbed, fib, lldp, lag_2, pfc_wd]
+            }
+'''
+
+### Default testbed file name
+TESTBED_FILE = 'testbed.csv'
+TESTCASE_FILE = 'roles/test/vars/testcases.yml'
+
+
+class ParseTestbedTopoinfo():
+    '''
+    Parse the CSV file used to describe whole testbed info
+    Please refer to the example of the CSV file format
+    CSV file first line is title
+    The topology name in title is using uniq-name | conf-name
+    '''
+    def __init__(self, testbed_file):
+        self.testbed_filename = testbed_file
+        self.testbed_topo = {}
+
+    def read_testbed_topo(self):
+        with open(self.testbed_filename) as f:
+            topo = csv.DictReader(f)
+            for line in topo:
+                tb_prop = {}
+                name = ''
+                for key in line:
+                    if ('uniq-name' in key or 'conf-name' in key) and '#' in line[key]:
+                        ### skip comment line
+                        continue
+                    elif 'uniq-name' in key or 'conf-name' in key:
+                        name = line[key]
+                    elif 'ptf_ip' in key and line[key]:
+                        ptfaddress = ipaddress.IPNetwork(line[key])
+                        tb_prop['ptf_ip'] = str(ptfaddress.ip)
+                        tb_prop['ptf_netmask'] = str(ptfaddress.netmask)
+                    else:
+                        tb_prop[key] = line[key]
+                if name:
+                    self.testbed_topo[name] = tb_prop
+        return
+
+    def get_testbed_info(self, testbed_name):
+        if testbed_name:
+            return self.testbed_topo[testbed_name]
```

Comment:
Probably it's better to return the same type from the function?
if we have testbed name - return a list with one element, if we don't have testbed name - return a list with all entries.